[6.7.0] Generated client imports .ts files as .js

### Bug description

```
Module not found: Can't resolve './enums.js'
```

the file is `enums.ts`, but the generated `client.ts` shows

```
import * as $Enums from "./enums.js"
```

for our own app, we don't use file extensions when importing, so that webpack can pick `js` or `ts` depending on environment

### Severity

⚠️ Major: Breaks core functionality (e.g., migrations fail)

### Reproduction

n/a

### Expected vs. Actual Behavior

expected: code to run
actual: did not run 

### Frequency

Consistently reproducible

### Does this occur in development or production?

Only in development (e.g., CLI tools, migrations, Prisma Studio)

### Is this a regression?

Yes, worked

### Workaround

<!-- e.g., "Downgrading to Prisma 5.1.0 fixes it" or "No workaround found" -->


### Prisma Schema & Queries

```prisma
// Add relevant schema.prisma snippet
```
```ts
// Add relevant Prisma Client query
```


### Prisma Config

```
previewFeatures = ["fullTextSearchPostgres", "nativeDistinct", "queryCompiler", "driverAdapters"]
```

### Logs & Debug Info

```
// Debug logs here
```


### Environment & Setup

next.js
webpack
prisma 6.7.0

### Prisma Version

```
6.7.0
```

```
platform:dev: [0] Module not found: Can't resolve './internal/class.js'
platform:dev: [0]   16 | import * as runtime from "@prisma/client/runtime/client"
platform:dev: [0]   17 | import * as $Enums from "./enums.js"
platform:dev: [0] > 18 | import * as $Class from "./internal/class.js"
platform:dev: [0]      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
platform:dev: [0]   19 | import * as Prisma from "./internal/prismaNamespace.js"
platform:dev: [0]   20 |
platform:dev: [0]   21 | export * as $Enums from './enums.js'
platform:dev: [0] 
platform:dev: [0] https://nextjs.org/docs/messages/module-not-found
platform:dev: [0] 
platform:dev: [0] ./apps/db/prisma/generated/client.ts:19:1
platform:dev: [0] Module not found: Can't resolve './internal/prismaNamespace.js'
platform:dev: [0]   17 | import * as $Enums from "./enums.js"
platform:dev: [0]   18 | import * as $Class from "./internal/class.js"
platform:dev: [0] > 19 | import * as Prisma from "./internal/prismaNamespace.js"
platform:dev: [0]      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
platform:dev: [0]   20 |
platform:dev: [0]   21 | export * as $Enums from './enums.js'
platform:dev: [0]   22 | /**
platform:dev: [0] 
platform:dev: [0] https://nextjs.org/docs/messages/module-not-found
platform:dev: [0] 
platform:dev: [0]  GET /api/users/me 500 in 53ms
platform:dev: [0]  ⨯ ./apps/db/prisma/generated/client.ts:21:1
platform:dev: [0] Module not found: Can't resolve './enums.js'
platform:dev: [0]   19 | import * as Prisma from "./internal/prismaNamespace.js"
platform:dev: [0]   20 |
platform:dev: [0] > 21 | export * as $Enums from './enums.js'
platform:dev: [0]      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
platform:dev: [0]   22 | /**
platform:dev: [0]   23 |  * ## Prisma Client
platform:dev: [0]   24 |  *
```

i'm not really sure how to make this more clear... the imports are all using the `.js` extension. the files are in the filesystem with `.ts` extensions

<img width="318" alt="Image" src="https://github.com/user-attachments/assets/48385016-2543-4139-ad55-de90739f77e6" />

  
-------
`'enums.js' !== 'enums.ts'`

--- a/packages/client-generator-ts/src/file-extensions.ts
+++ b/packages/client-generator-ts/src/file-extensions.ts
@@ -99,11 +99,9 @@ function inferImportFileExtensionFromTsConfig(
 
   // Otherwise, we must either use the `.js`/`.mjs`/`.cjs` extension, or none at
   // all. The latter will only work when using a bundler, or when targeting
-  // CommonJS. Using a JavaScript extension is safer because it will always work
-  // when transpiling but may be confusing to some users, especially when using
-  // a bundler. Therefore we conservatively use bare module specifiers without
-  // an extension when we are absolutely sure it is supported, and fall back to
-  // the matching JavaScript extension otherwise.
+  // CommonJS. We use bare module specifiers without an extension when we are
+  // sure it is supported, and fall back to the matching JavaScript extension
+  // otherwise.
 
   const moduleResolution = tsconfig.compilerOptions?.moduleResolution?.toLowerCase() as
     | TsConfigJson.CompilerOptions.ModuleResolution
@@ -111,7 +109,7 @@ function inferImportFileExtensionFromTsConfig(
 
   const module = tsconfig.compilerOptions?.module?.toLowerCase() as TsConfigJson.CompilerOptions.Module | undefined
 
-  if (moduleResolution !== undefined && moduleResolution !== 'classic' && module === 'commonjs') {
+  if (module === 'commonjs' || moduleResolution === 'bundler') {
     return ''
   }
 

--- a/packages/client/tests/e2e/ts-client/node-tsup-esm/.gitignore
+++ b/packages/client/tests/e2e/ts-client/node-tsup-esm/.gitignore
@@ -1,0 +1,3 @@
+/src/generated
+/prisma/db
+/dist

--- a/packages/client/tests/e2e/ts-client/node-tsup-esm/_steps.cts
+++ b/packages/client/tests/e2e/ts-client/node-tsup-esm/_steps.cts
@@ -1,0 +1,21 @@
+import { $ } from 'zx'
+
+import { executeSteps } from '../../_utils/executeSteps'
+
+void executeSteps({
+  setup: async () => {
+    await $`pnpm install`
+    await $`pnpm prisma generate`
+    await $`pnpm prisma db push --force-reset --skip-generate`
+    await $`pnpm tsup src/test.ts --format esm`
+  },
+  test: async () => {
+    await $`pnpm exec prisma -v`
+    await $`node dist/test.js`
+    await $`pnpm exec tsc --noEmit`
+  },
+  finish: async () => {
+    await $`echo "done"`
+  },
+  // keep: true, // keep docker open to debug it
+})

--- a/packages/client/tests/e2e/ts-client/node-tsup-esm/package.json
+++ b/packages/client/tests/e2e/ts-client/node-tsup-esm/package.json
@@ -1,0 +1,14 @@
+{
+  "private": true,
+  "version": "0.0.0",
+  "type": "module",
+  "scripts": {},
+  "dependencies": {
+    "@prisma/client": "/tmp/prisma-client-0.0.0.tgz"
+  },
+  "devDependencies": {
+    "@types/node": "18.19.76",
+    "prisma": "/tmp/prisma-0.0.0.tgz",
+    "tsup": "8.4.0"
+  }
+}

--- a/packages/client/tests/e2e/ts-client/node-tsup-esm/prisma/schema.prisma
+++ b/packages/client/tests/e2e/ts-client/node-tsup-esm/prisma/schema.prisma
@@ -1,0 +1,36 @@
+// This is your Prisma schema file,
+// learn more about it in the docs: https://pris.ly/d/prisma-schema
+
+generator client {
+  provider = "prisma-client"
+  output   = "../src/generated/prisma"
+}
+
+datasource db {
+  provider = "sqlite"
+  url      = "file:/tmp/prisma.db"
+}
+
+model User {
+  id      Int      @id @default(autoincrement())
+  email   String   @unique
+  name    String?
+  posts   Post[]
+  profile Profile?
+}
+
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  content   String?
+  published Boolean @default(false)
+  author    User    @relation(fields: [authorId], references: [id])
+  authorId  Int
+}
+
+model Profile {
+  id     Int     @id @default(autoincrement())
+  bio    String?
+  user   User    @relation(fields: [userId], references: [id])
+  userId Int     @unique
+}

--- a/packages/client/tests/e2e/ts-client/node-tsup-esm/readme.md
+++ b/packages/client/tests/e2e/ts-client/node-tsup-esm/readme.md
@@ -1,0 +1,1 @@
+TypeScript client, ESM, bundled with tsup.

--- a/packages/client/tests/e2e/ts-client/node-tsup-esm/src/test.ts
+++ b/packages/client/tests/e2e/ts-client/node-tsup-esm/src/test.ts
@@ -1,0 +1,15 @@
+import assert from 'node:assert/strict'
+
+import { PrismaClient } from './generated/prisma/client'
+
+const prisma = new PrismaClient()
+
+await prisma.user.deleteMany()
+
+const user = await prisma.user.create({
+  data: { email: 'john@doe.io' },
+})
+
+console.log(user)
+
+assert.equal(await prisma.user.count(), 1)

--- a/packages/client/tests/e2e/ts-client/node-tsup-esm/tsconfig.json
+++ b/packages/client/tests/e2e/ts-client/node-tsup-esm/tsconfig.json
@@ -1,0 +1,10 @@
+{
+  "exclude": ["_steps.cts"],
+  "compilerOptions": {
+    "target": "ESNext",
+    "module": "ESNext",
+    "moduleResolution": "bundler",
+    "strict": true,
+    "skipLibCheck": false
+  }
+}

